confluent_kafka/src/confluent_kafka.c

```diff
@@ -585,7 +612,7 @@ PyObject *Message_new0 (const Handle *handle, const rd_kafka_message_t *rkm) {
 	self->timestamp = rd_kafka_message_timestamp(rkm, &self->tstype);
 
     if (!rd_kafka_message_headers(rkm, &hdrs)) {
-        self->headers = c_headers_to_py(hdrs);
+        self->c_headers = rd_kafka_headers_copy(hdrs);
```

Comment:
We can take ownership of the headers by doing `rd_kafka_message_set_headers(rkm, NULL)`, to avoid copying them.

Reply:
Ok i think i follow. We can store the address of the existing c_headers on the python Message struct. Then we disassociate the headers from rkm so that when it's destroyed the underlying headers aren't.

When I make this change I am getting a warning from the compiler that the `rkm` is declared as a const but we are mutating it. Should i change the signature to match?

Reply:
Right on!

Yes, remove const from Message_new0()

Reply:
I am not doing something right. If I ignore the warning the headers are still destroyed.

https://github.com/edenhill/librdkafka/blob/master/src/rdkafka_msg.c#L1061

`rd_kafka_message_set_headers` calls `rd_kafka_headers_destroy` on the existing headers. I am reading that right?

Reply:
You are absolutely right, sorry for the confusion.
Let me add a `rd_kafka_message_detach_headers(rkm, &hdrs)` API for this purpose.

Reply:
There is another issue with removing the const from the `Message_new0`. This is used by the delivery callback and that has a signature of const too. 

So you end up with:

```
  In file included from confluent_kafka/src/confluent_kafka.h:21:0,
                   from confluent_kafka/src/Producer.c:17:
  /home/vagrant/repos/librdkafka/out/include/librdkafka/rdkafka.h:1326:6: note: expected ‘void (*)(struct rd_kafka_t *, const struct rd_kafka_message_t *, void *)’ but argument is of type ‘void (*)(struct rd_kafka_t *, struct rd_kafka_message_t *, void *)’
   void rd_kafka_conf_set_dr_msg_cb(rd_kafka_conf_t *conf,
```